docs/auth_server.rst

```diff
@@ -0,0 +1,146 @@
+Authentication Server
+=====================
+
+Superdesk provides an OAuth2 authentication server for the production API. The server is
+using the `Client Credentials Grant`_ with `JWT authentication`_.
+
+The endpoint used in Superdesk is ``/api/auth_server/token``.
+
+.. _Client Credentials Grant: https://tools.ietf.org/html/rfc6749#section-4.4
+.. _JWT authentication: https://tools.ietf.org/html/rfc7519
+
+Overview
+--------
+
+Here is the sequence of the authentication:
+
+.. uml::
+
+    title Authentication server flow in Superdesk (for third party clients)
+
+    boundary client
+    control "authentication server" as auth_server
+    database "data backend\n(//Mongo DB//)" as data_backend
+    boundary "resource owner\n(//Production API//)" as res_owner
+
+    client -> auth_server: request token
+    note right
+        using **client credentials** grant
+        and **JWT**
+    end note
+
+    alt success
+        auth_server -> data_backend: create token
+        client <- auth_server : access token
+        loop
+            client -> res_owner: request with access token
+            client <- res_owner: request answer
+        end
+
+    else failure
+        auth_server -> client: HTTP error\n(400 or something else)
+        note right
+            a //JSON// payload with ""error"" key should be present,
+            the value being an explanation of the issue
+        end note
+
+    end
+
+Allowing a client
+-----------------
+
+A client needs to be put in Superdesk's ``settings.py`` to be authorized to have an access
+token.
+
+The clients are put in the ``AUTH_SERVER_CLIENTS`` parameters which is a Python list
+of dictionaries. Each dictionary MUST have the following keys:
+
+======  ===============  ======================================================
+key     type             explanation
+======  ===============  ======================================================
+id      string           unique identifier of the client
+secret  string           shared secret between the client and the auth server
+scope   list of strings  list of allowed privileges for this client (see below)
+======  ===============  ======================================================
+
+Here is an example of client configuration:
+
+.. code:: python
+
+    AUTH_SERVER_CLIENTS = [
+        {
+            "id": "123456",
+            "secret": "789101112",
```

Comment:
imo we shouldn't store the secret in plain text on server, and ideally this would be in db (not plain text again)

Reply:
I was thinking that this could be a first step, and later we can add a client UI for that with a proper API and storage in db.

An other option would be to create a `manage.py` command for that, should I go this way?

Reply:
Also I don't see how this could be stored encrypted in db without a masterkey or something like that. We can't store only a hash, the secret is need by authlib to validate the signature.

Reply:
isn't it only used here? https://github.com/superdesk/superdesk-core/pull/1674/files#diff-bcd8f2afa533368cd035ae88d6b012c6R41

Reply:
no, it's also used by JWT for signing auth token.